sdks/python/apache_beam/io/filebasedsource.py

```diff
@@ -95,12 +104,7 @@ def __init__(self,
       raise TypeError('compression_type must be CompressionType object but '
                       'was %s' % type(compression_type))
     self._compression_type = compression_type
-    if compression_type in (CompressionTypes.UNCOMPRESSED,
-                            CompressionTypes.AUTO):
-      self._splittable = splittable
-    else:
-      # We can't split compressed files efficiently so turn off splitting.
-      self._splittable = False
+    self._splittable = splittable
```

Comment:
<!--new_thread; commit:64a86ed2bb30b5a9fc7127bc2c86deccb4f7a65a; resolved:0-->
Do you need to also check `_determine_splittability_from_compression_type` ?